src/storage_broker/app.py

```diff
@@ -102,7 +102,11 @@ def main():
     producer = produce.init_producer()
 
     while running:
+        if event and event.is_set():
```

Comment:
I wonder if it would be possible to just replace the "running" boolean with the event object.  Just a thought/idea.  I haven't looked into that part of it in great detail.

Reply:
Great idea! Replaced the running boolean with event object. It was quite simple to do :+1: